tools/schemapi/schemapi.py

```diff
@@ -741,6 +742,18 @@ def __repr__(self) -> str:
 
 
 Undefined = UndefinedType()
+T = TypeVar("T")
+Optional: TypeAlias = Union[T, UndefinedType]
```

Comment:
This is nice!

Reply:
As a comparison, see the gradual improvements in `MarkMethodMixin.mark_arc`:

- [main](https://github.com/vega/altair/blob/18a2c3c237014591d172284560546a2f0ac1a883/altair/vegalite/v5/schema/mixins.py#L22)
- [`|` Operator introduced](https://github.com/vega/altair/blob/36da1c79ba5aacfa4a8e176668b22ce7d647b92e/altair/vegalite/v5/schema/mixins.py#L27)
- [`Optional` introduced](https://github.com/vega/altair/blob/fbc02e2d577a46509436ae6f40b9b56035e996c8/altair/vegalite/v5/schema/mixins.py#L31)

There's still some low-hanging fruit with a future addition aliases for all the long `Literal` types, but even as-is, these changes have reduced ~9000 lines.

Reply:
> There's still some low-hanging fruit with a future addition aliases for all the long `Literal` types, but even as-is, these changes have reduced ~9000 lines.

`Literal` -> `TypeAlias` added in [`a7b0ab5` (#3431)](https://github.com/vega/altair/pull/3431/commits/a7b0ab58b5f0e24b996d44f3e81b5235600defa1)

Reply:
I do like the use of a `TypeVar` for this but I don't know if we should name it `Optional` as it already has a different meaning which is well known to users. If I read `Optional` in my IDE, I expect that I can pass `None` and I might write code for it. What do you think? I can't come up with a better name for it right now...

I really like the literals to reduce the code!